src/main/webapp/app/iris/websocket.service.ts

```diff
@@ -0,0 +1,78 @@
+import { IrisHttpMessageService } from 'app/iris/http-message.service';
+import { Observable, Subscription } from 'rxjs';
+import { User } from 'app/core/user/user.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { Injectable, OnDestroy } from '@angular/core';
+import { JhiWebsocketService } from 'app/core/websocket/websocket.service';
+import { IrisMessageStore } from 'app/iris/message-store.service';
+import { ActiveConversationMessageLoadedAction, MessageStoreState } from 'app/iris/message-store.model';
+import { IrisMessageDescriptor, IrisSender } from 'app/entities/iris/iris.model';
+
+@Injectable()
+export class IrisWebsocketService implements OnDestroy {
+    private user: User;
+    private sessionId: number;
+    private subscriptionChannel?: string;
+    private connectionStateSub: Subscription;
+
+    constructor(
+        protected messageService: IrisHttpMessageService,
+        protected accountService: AccountService,
+        private jhiWebsocketService: JhiWebsocketService,
+        private messageStore: IrisMessageStore,
+    ) {
+        this.accountService.identity().then((user: User) => {
+            this.user = user!;
+        });
+        this.connectionStateSub = this.jhiWebsocketService.connectionState.subscribe((newState) => {
+            //TODO
+        });
+    }
+
+    get messages(): Observable<MessageStoreState> {
+        return this.messageStore.getState();
+    }
+
+    ngOnDestroy(): void {
+        if (this.subscriptionChannel) {
+            this.jhiWebsocketService.unsubscribe(this.subscriptionChannel);
+        }
+        this.connectionStateSub.unsubscribe();
+    }
+
+    getUser(): User {
+        return this.user;
+    }
+
+    /**
+     * Creates (and updates) the websocket channel for receiving messages in dedicated channels;
+     * @param channel which the iris service should subscribe to
+     */
+    changeWebsocketSubscription(channel: string | null): void {
```

Comment:
I think the channel should be created automatically in this method using a session as an input parameter